backend/app/adapter/sqldb/user_short_link.go

```diff
@@ -63,6 +63,26 @@ func (u UserShortLinkSQL) FindAliasesByUser(user entity.User) ([]string, error)
 	return aliases, nil
 }
 
+// HasMapping checks whether a given short link belongs to a user.
+func (u UserShortLinkSQL) HasMapping(user entity.User, alias string) (bool, error) {
+	query := fmt.Sprintf(`SELECT "%s" FROM "%s" WHERE "%s"=$1 AND "%s"=$2`,
+		table.UserShortLink.ColumnUserID,
+		table.UserShortLink.TableName,
+		table.UserShortLink.ColumnUserID,
+		table.UserShortLink.ColumnShortLinkAlias,
+	)
+
+	var id string
+	err := u.db.QueryRow(query, user.ID, alias).Scan(&id)
+	if err == sql.ErrNoRows {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
```

Comment:
I wonder whether we can do the following:
```go
return true, err
```
When there is err, the return value is ignored.

Reply:
Should the value be trusted if there was an error?

Reply:
I see what you mean now. I think it would still be better to have it return as is in case someone accidentally uses the returned value.

Reply:
@oatovar It's Go's convention not to use the returned value if there is error. I am fine keeping the code as you have.